SIT.Manager.Avalonia/ViewModels/ToolsPageViewModel.cs

```diff
@@ -217,4 +236,115 @@ private async Task ClearCache()
+        }
+
+        if (results.IncludeServerLog && !string.IsNullOrEmpty(_configService.Config.AkiServerPath))
+        {
+            DirectoryInfo serverLogDirectory = new(Path.Combine(_configService.Config.AkiServerPath, "user", "logs"));
+            if (serverLogDirectory.Exists)
+            {
+                IEnumerable<FileInfo> files = serverLogDirectory.GetFiles("*.log");
+                files = files.OrderBy(x => x.LastWriteTime);
+                if (files.Any())
+                {
+                    string serverLogFile = files.First().FullName;
+                    DiagnosticData.Add(new(Path.GetFileName(serverLogFile), await File.ReadAllTextAsync(serverLogFile)));
+                }
+            }
+        }
+
+        if (results.IncludeDiagnosticLog)
+        {
+            //TODO: Add more diagnostics if needed
+            StringBuilder sb = new("#--- DIAGNOSTICS LOG ---#\n\n");
+
+            //Get all networks adaptors local address if they're online
+            sb.AppendLine("#-- Network Information: --#\n");
+            foreach (NetworkInterface networkInterface in NetworkInterface.GetAllNetworkInterfaces())
+            {
+                if (networkInterface.OperationalStatus == OperationalStatus.Up)
+                {
+                    foreach (UnicastIPAddressInformation ip in networkInterface.GetIPProperties().UnicastAddresses)
+                    {
+                        if (ip.Address.AddressFamily == System.Net.Sockets.AddressFamily.InterNetwork)
+                        {
+                            sb.AppendLine($"Network Interface: {networkInterface.Name}");
+                            sb.AppendLine($"Interface Type: {networkInterface.NetworkInterfaceType.ToString()}");
+                            sb.AppendLine($"Address: {ip.Address}\n");
+                        }
+                    }
+                }
+            }
+
+            //TODO: Add system hardware information
+
+            DiagnosticData.Add(new("Diagnostics.log", sb.ToString()));
+        }
+
+        if (results.IncludeHttpJson)
+        {
+            FileInfo httpJsonFile = new(Path.Combine(_configService.Config.AkiServerPath, "Aki_Data", "Server", "configs", "http.json"));
+            if (httpJsonFile.Exists)
+                DiagnosticData.Add(new(httpJsonFile.Name, await File.ReadAllTextAsync(httpJsonFile.FullName)));
+        }
+
+        //I hate doing it this way but at least icanhazip is owned by cloudlfare
+        HttpResponseMessage resp = await _httpClient.GetAsync("https://ipv4.icanhazip.com/");
+        string externalAddress = await resp.Content.ReadAsStringAsync();
+
+        List<Tuple<string, string>> CleanLogFiles = new(DiagnosticData.Count);
+        foreach (Tuple<string, string> diagnosticFile in DiagnosticData)
+        {
+            CleanLogFiles.Add(new(diagnosticFile.Item1, diagnosticFile.Item2.Replace(externalAddress.Trim(), "REACTED")));
```

Comment:
One other point that's very minor is you could just do the trim once on external address rather than everytime in the foreach look